services/src/main/java/org/keycloak/services/resources/LoginActionsService.java

```diff
@@ -945,7 +950,25 @@ private String getActionCookie() {
     public static void createActionCookie(RealmModel realm, UriInfo uriInfo, ClientConnection clientConnection, String sessionId) {
         CookieHelper.addCookie(ACTION_COOKIE, sessionId, AuthenticationManager.getRealmCookiePath(realm, uriInfo), null, null, -1, realm.getSslRequired().isRequired(clientConnection), true);
     }
-
+    
+    private static void isForcePasswordUpdateRequired(RealmModel realm, UserModel user) {
```

Comment:
This should be invoked from nextActionAfterAuthentication on L378 (before isTotpConfigurationRequired is invoked).